src/app/components/history/history.component.html

```diff
@@ -1,19 +1,36 @@
 <div class="container-fluid">
+  <div class="row mb-2">
+    <div class="btn-group btn-group-toggle">
+      <label class="btn btn-secondary" for="filter_all" [class.active]="filter === 'all'">
+        <input type="radio" [ngModel]="filter" value="all" name="options" id="filter_all" (ngModelChange)="filterHistory($event)" autocomplete="off"> {{ 'All' | translate }}
+      </label>
+      <label class="btn btn-secondary" for="filter_delegated" [class.active]="filter === 'delegated'">
+        <input type="radio" [ngModel]="filter" value="delegated" name="filter" id="filter_delegated" (ngModelChange)="filterHistory($event)" autocomplete="off"> {{ 'Delegated' | translate }}
+      </label>
+      <label class="btn btn-secondary" for="filter_undelegated" [class.active]="filter === 'undelegated'">
+        <input type="radio" [ngModel]="filter" value="undelegated" name="filter" id="filter_undelegated" (ngModelChange)="filterHistory($event)" autocomplete="off"> {{ 'Undelegated' | translate }}
+      </label>
+    </div>
+  </div>
+
   <div class="row">
-    <div *ngIf="history.length > 0;else noHistory" class="list-group w-100">
+    <div *ngIf="filteredHistory.length > 0;else noHistory" class="list-group w-100">
       <div *ngFor="let item of historyItems" class="list-group-item list-group-item-action list-group-item-{{item.color}}">
-        <a routerLink="/result/{{item.id}}"> {{ item.creation_time | amFromUtc | date:'yyyy-MM-dd HH:mm zzzz' }}</a>
-
+        <div class="d-flex w-100 justify-content-between">
+          <a routerLink="/result/{{item.id}}"> {{ item.creation_time | amFromUtc | date:'yyyy-MM-dd HH:mm zzzz' }}</a>
+          <small *ngIf="item.undelegated" class="text-muted">{{ "Undelegated" | translate }}</small>
```

Comment:
Change from `{{ "Undelegated" | translate }}` to `{{ 'UNDELEGATED_SINGULAR' | translate }}`

I do not find where DELEGATED_SINGULAR should be.